FLT/Mathlib/Algebra/Order/GroupWithZero.lean

```diff
@@ -0,0 +1,32 @@
+import Mathlib.Algebra.Order.GroupWithZero.Canonical
+
+namespace WithZero
+
+lemma exists_ne_zero_and_lt {G : Type*} [LinearOrder G] [NoMinOrder G]
+  {x : WithZero G} (hx : x ≠ 0) :
+    ∃ y, y ≠ 0 ∧ y < x := by
+  obtain ⟨z, hlt⟩ := exists_lt (WithZero.unzero hx)
+  rw [← WithZero.coe_lt_coe, WithZero.coe_unzero hx] at hlt
+  use z, WithZero.coe_ne_zero
+
+lemma exists_ne_zero_and_le_and_le {G : Type*} [LinearOrder G]
+  {x y : WithZero G} (hx : x ≠ 0) (hy : y ≠ 0) :
+    ∃ z, z ≠ 0 ∧ z ≤ x ∧ z ≤ y := by
+  let z := x ⊓ y
+  use z
+  constructor
+  . unfold z
+    intro hz
+    rw [min_eq_iff] at hz
+    tauto
+  . constructor <;> simp_all [z]
```

Comment:
```suggestion
  use x ⊓ y
  refine ⟨fun hz ↦ ?_, by simp, by simp⟩
  rw [min_eq_iff] at hz
  tauto
```
No different, just noting some techniques: you don't need to introduce z, and for P \and Q \and R goals it's probably easier to use `refine \<?_, ?_, ?_\>` than two `constructors.